service/history/commandChecker.go

```diff
@@ -644,6 +646,14 @@ func (v *commandAttrValidator) validatedTaskQueue(
 	return taskQueue, nil
 }
 
+func (v *commandAttrValidator) validateActivityRetryPolicy(attributes *commandpb.ScheduleActivityTaskCommandAttributes) error {
+	if attributes.RetryPolicy == nil {
+		attributes.RetryPolicy = v.defaultActivityRetryPolicy
+	}
+
+	return common.ValidateRetryPolicy(attributes.RetryPolicy)
```

Comment:
I believe it is better to validate default policy just once (in `fromConfigToActivityRetryPolicy`) and skip validation here in case if we are using default policy.

Reply:
sure, will incorporate that in my change